Sources/SafeDICore/Generators/DependencyTreeGenerator.swift

```diff
@@ -296,11 +296,17 @@ public final class DependencyTreeGenerator {
                 let parentContainsProperty = receivableProperties.contains(receivedProperty)
                 let propertyIsCreatedAtThisScope = createdProperties.contains(receivedProperty)
                 if !parentContainsProperty && !propertyIsCreatedAtThisScope {
-                    unfulfillableProperties.insert(.init(
-                        property: receivedProperty,
-                        instantiable: scope.instantiable,
-                        parentStack: instantiables.elements)
-                    )
+                    if instantiables.elements.isEmpty {
+                        // This property's scope is not a real root instantiable! Remove it from the list.
+                        rootInstantiableTypes.remove(scope.instantiable.concreteInstantiableType)
```

Comment:
a little bit of hax, but when we're validating properties and we find a root that isn't really a root... just remove it.